sdk/src/jupiter/jupiterClientV6.ts

```diff
@@ -0,0 +1,357 @@
+	 */
+	public async getQuote({
+		inputMint,
+		outputMint,
+		amount,
+		slippageBps = 50,
+		swapMode = 'ExactIn',
+		onlyDirectRoutes = false,
+	}: {
+		inputMint: PublicKey;
+		outputMint: PublicKey;
+		amount: BN;
+		slippageBps?: number;
+		swapMode?: SwapMode;
+		onlyDirectRoutes?: boolean;
+	}): Promise<QuoteResponse> {
+		const params = new URLSearchParams({
+			inputMint: inputMint.toString(),
+			outputMint: outputMint.toString(),
+			amount: amount.toString(),
+			slippageBps: slippageBps.toString(),
+			swapMode,
+			onlyDirectRoutes: onlyDirectRoutes.toString(),
+		}).toString();
+		const quote = await (
+			await fetch(`https://quote-api.jup.ag/v6/quote?${params}`)
+		).json();
+		return quote;
+	}
+
+	/**
+	 * Get a swap transaction for quote
+	 * @param quoteResponse quote to perform swap
+	 * @param userPublicKey the signer's wallet public key
+	 * @param slippageBps the slippage tolerance in basis points
+	 */
+	public async getSwapTransaction({
+		quote,
+		userPublicKey,
+		slippageBps = 50,
+	}: {
+		quote: QuoteResponse;
+		userPublicKey: PublicKey;
+		slippageBps?: number;
+	}): Promise<VersionedTransaction> {
+		const resp = await (
+			await fetch(`${this.url}/swap`, {
+				method: 'POST',
+				headers: {
+					'Content-Type': 'application/json',
+				},
+				body: JSON.stringify({
+          quoteResponse: quote,
+					userPublicKey,
+					slippageBps,
+				}),
+			})
+		).json();
+		const { swapTransaction } = resp;
+
+		const swapTransactionBuf = Buffer.from(swapTransaction, 'base64');
+		return VersionedTransaction.deserialize(swapTransactionBuf);
+	}
+
+	/**
+	 * Get the transaction message and lookup tables for a transaction
+	 * @param transaction
+	 */
+	public async getTransactionMessageAndLookupTables({
+		transaction,
+	}: {
+		transaction: VersionedTransaction;
+	}): Promise<{
+		transactionMessage: TransactionMessage;
+		lookupTables: AddressLookupTableAccount[];
+	}> {
+		const message = transaction.message;
+
+		const lookupTables = (
+			await Promise.all(
+				message.addressTableLookups.map(async (lookup) => {
+					return await this.getLookupTable(lookup.accountKey);
+				})
+			)
+		).filter((lookup) => lookup);
+
+		const transactionMessage = TransactionMessage.decompile(message, {
+			addressLookupTableAccounts: lookupTables,
+		});
+
+		return {
+			transactionMessage,
+			lookupTables,
+		};
+	}
+
+	async getLookupTable(
+		accountKey: PublicKey
+	): Promise<AddressLookupTableAccount> {
+		if (this.lookupTableCahce.has(accountKey.toString())) {
+			return this.lookupTableCahce.get(accountKey.toString());
+		}
+
+		return (await this.connection.getAddressLookupTable(accountKey)).value;
+	}
+
+	/**
+	 * Get the jupiter instructions from transaction by filtering out instructions to compute budget and associated token programs
+	 * @param transactionMessage the transaction message
+	 * @param inputMint the input mint
+	 * @param outputMint the output mint
+	 */
+	public getJupiterInstructions({
```

Comment:
if may be better to use [/swap-transactions](https://station.jup.ag/docs/v6-beta/swap-api#instructions-instead-of-transaction) for v6 instead of building it from ourselves like in v4